src/types/app_config.go

```diff
@@ -39,6 +39,14 @@ func NewAppConfig(appDir string) (result AppConfig, err error) {
 	return result, result.validateAppConfig()
 }
 
+// AddService adds serviceRole to the appConfig object
+func (a AppConfig) AddService(serviceRole string) {
```

Comment:
This doesn't update the config because this function is passed a copy instead of a reference. We should update all of these to be pointer receivers and have NewAppConfig return a pointer. Please do that in a separate PR that this one depends on.

https://tour.golang.org/methods/4